e2e-tests/desktop/tests/targets.spec.js

```diff
@@ -4,19 +4,228 @@
  */
 
 import { expect, test } from '../fixtures/baseTest.js';
+import * as boundaryHttp from '../../helpers/boundary-http';
+
+const hostName = 'Host name for test';
+let org;
+let targetWithHost;
+let targetWithTwoHosts;
+let sshTarget;
+let credential;
+
+test.beforeAll(
+  async ({ request, targetAddress, targetPort, sshUser, sshKeyPath }) => {
+    org = await boundaryHttp.createOrg(request);
+    const project = await boundaryHttp.createProject(request, org.id);
+
+    // Create host set
+    const hostCatalog = await boundaryHttp.createStaticHostCatalog(
+      request,
+      project.id,
+    );
+    const host = await boundaryHttp.createHost(request, {
+      hostCatalogId: hostCatalog.id,
+      address: targetAddress,
+    });
+    const host2 = await boundaryHttp.createHost(request, {
+      hostCatalogId: hostCatalog.id,
+      address: targetAddress,
+      name: hostName,
+    });
+
+    const hostSet = await boundaryHttp.createHostSet(request, hostCatalog.id);
+    await boundaryHttp.addHostToHostSet(request, {
+      hostSet,
+      hostIds: [host.id],
+    });
+
+    // Create a host set with 2 hosts
+    let hostSetWithTwoHosts = await boundaryHttp.createHostSet(
+      request,
+      hostCatalog.id,
+    );
+    hostSetWithTwoHosts = await boundaryHttp.addHostToHostSet(request, {
+      hostSet: hostSetWithTwoHosts,
+      hostIds: [host.id],
+    });
+    hostSetWithTwoHosts = await boundaryHttp.addHostToHostSet(request, {
+      hostSet: hostSetWithTwoHosts,
+      hostIds: [host2.id],
+    });
+
+    // Create a static credential store and key pair credential
+    const credentialStore = await boundaryHttp.createStaticCredentialStore(
+      request,
+      project.id,
+    );
+    credential = await boundaryHttp.createStaticCredentialKeyPair(request, {
+      credentialStoreId: credentialStore.id,
+      username: sshUser,
+      sshKeyPath,
+    });
+
+    // Create tcp target with host set and 1 host
+    targetWithHost = await boundaryHttp.createTarget(request, {
+      scopeId: project.id,
+      type: 'tcp',
+      port: targetPort,
+    });
+    targetWithHost = await boundaryHttp.addHostSource(request, {
+      target: targetWithHost,
+      hostSourceIds: [hostSet.id],
+    });
+    targetWithHost = await boundaryHttp.addBrokeredCredentials(request, {
+      target: targetWithHost,
+      credentialIds: [credential.id],
+    });
+
+    // Create tcp target with host set and 2 hosts
+    targetWithTwoHosts = await boundaryHttp.createTarget(request, {
+      scopeId: project.id,
+      type: 'tcp',
+      port: targetPort,
+    });
+    targetWithTwoHosts = await boundaryHttp.addHostSource(request, {
+      target: targetWithTwoHosts,
+      hostSourceIds: [hostSetWithTwoHosts.id],
+    });
+    targetWithTwoHosts = await boundaryHttp.addBrokeredCredentials(request, {
+      target: targetWithTwoHosts,
+      credentialIds: [credential.id],
+    });
+
+    // Create an SSH target with address
```

Comment:
Are we going to assume these tests are always going to run against an enterprise boundary instance?

Reply:
Yeah, that's one other thing I wanted to discuss, as I haven't added a README yet. It assumes right now it's only against an enterprise controller since SSH targets (and some of the vault setup) are pretty important that always needs to be tested which is why I didn't add any tags like in admin and anything we test in CE should work the same in ent.

I'm not sure how much value there is for testing against a CE controller right now but maybe it's something we can split out down the line with tagging if it's needed?

Reply:
Yea that makes sense. Just wanted to make sure there weren't any major differences.